api-participant/src/main/java/bio/terra/pearl/api/participant/controller/file/ParticipantFileController.java

```diff
@@ -0,0 +1,97 @@
+package bio.terra.pearl.api.participant.controller.file;
+
+import bio.terra.pearl.api.participant.api.ParticipantFileApi;
+import bio.terra.pearl.api.participant.service.RequestUtilService;
+import bio.terra.pearl.api.participant.service.file.ParticipantFileExtService;
+import bio.terra.pearl.core.model.file.ParticipantFile;
+import bio.terra.pearl.core.model.participant.ParticipantUser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.http.HttpServletRequest;
+import java.io.InputStream;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.multipart.MultipartFile;
+
+@Controller
+public class ParticipantFileController implements ParticipantFileApi {
+  private final ParticipantFileExtService participantFileExtService;
+  private final HttpServletRequest request;
+  private final ObjectMapper objectMapper;
+  private final RequestUtilService requestUtilService;
+
+  public ParticipantFileController(
+      ParticipantFileExtService participantFileExtService,
+      HttpServletRequest request,
+      ObjectMapper objectMapper,
+      RequestUtilService requestUtilService) {
+    this.participantFileExtService = participantFileExtService;
+    this.request = request;
+    this.objectMapper = objectMapper;
+    this.requestUtilService = requestUtilService;
+  }
+
+  @Override
+  public ResponseEntity<Resource> download(
+      String portalShortcode,
+      String envName,
+      String studyShortcode,
```

Comment:
Oh also, I think we should do a check to verify that the participant is in this study env